common/src/main/java/com/wynntils/screens/base/widgets/ItemSearchWidget.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.base.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.screens.base.TextboxScreen;
+import com.wynntils.services.itemfilter.SearchQuery;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import java.util.function.Consumer;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.gui.components.Tooltip;
+import net.minecraft.network.chat.Component;
+
+public class ItemSearchWidget extends SearchWidget {
+    private SearchQuery searchQuery;
+
+    private Consumer<SearchQuery> onSearchQueryUpdateConsumer;
+
+    public ItemSearchWidget(
+            int x,
+            int y,
+            int width,
+            int height,
+            Consumer<SearchQuery> onSearchQueryUpdateConsumer,
+            TextboxScreen textboxScreen) {
+        super(x, y, width, height, null, textboxScreen);
+        this.onSearchQueryUpdateConsumer = onSearchQueryUpdateConsumer;
+        this.searchQuery = SearchQuery.fromQueryString("");
+    }
+
+    @Override
+    protected void renderText(
+            PoseStack poseStack,
+            String renderedText,
+            int renderedTextStart,
+            String firstPortion,
+            String highlightedPortion,
+            String lastPortion,
+            Font font,
+            int firstWidth,
+            int highlightedWidth,
+            int lastWidth,
+            boolean defaultText) {
+        if (defaultText || !getTextBoxInput().equals(searchQuery.getQueryString())) {
```

Comment:
`!getTextBoxInput().equals(searchQuery.getQueryString())` If our parsed query is not the same as the input, do we just render the default text? Why? Can this happen?

Reply:
If `defaultText`, then we let the superclass handle it. (It should display the default text or whatever, this does not relate to our specific implementation)

The parsed query should never be the same as the input, since it is updated in `onUpdate()`. If for a mysterious reason this happens, I thought it was better to let the default SearchWidget renering behavious (i.e. rendering the text in plain white), rather than messed up colouring not matching the string displayed.

Reply:
Your comment contradicts the code. You mean that the parsed query should always be the same as the input? Otherwise, the test in the code is wrong.

Reply:
@magicus My apologies, the parsed query should indeed always be the same as the input.

Reply:
@kristofbolyai Does it seem good to you ? I understand that it may be wrong to delegate to the superclass in case of an unexpected state (i.e. query != input), but I could not figure out a better way. It's either rendering it with wrong syntax colouring or no colouring at all.